test/decimal.test.js

```diff
@@ -63,4 +69,431 @@ describe('model with decimal property', function() {
+        })
+        .then(function(createdInstance) {
+          createdInstance.imdb.rating.should.be.instanceOf(Decimal128);
+          createdInstance.imdb.rating.should.deepEqual(Decimal128.fromString('4.5'));
+          createdInstance.imdb.innerObj.innerObj.innerObj.innerObj.testDecimal
+            .should.be.instanceOf(Decimal128);
+          createdInstance.imdb.innerObj.innerObj.innerObj.innerObj.testDecimal
+            .should.deepEqual(Decimal128.fromString('55.55'));
+          createdInstance.imdb.innerArray[0].testDecimal
+            .should.be.instanceOf(Decimal128);
+          createdInstance.imdb.innerArray[0].testDecimal
+            .should.deepEqual(Decimal128.fromString('99.55'));
+          createdInstance.imdb.innerArray[0].ObjInsideAnArray.testDecimal
+            .should.deepEqual(Decimal128.fromString('99.55'));
+          createdInstance.imdb.innerArray[0].ObjInsideAnArray.testDecimal
+            .should.be.instanceOf(Decimal128);
+          createdInstance.imdb.innerArray[0].nestedArray[0]
+            .testDecimal.should.be.instanceOf(Decimal128);
+          createdInstance.imdb.innerArray[0].nestedArray[0]
+            .testDecimal.should.deepEqual(Decimal128.fromString('99.55'));
+          return modelWithDeepNestedDecimalProps.updateAll({id: instanceId}, updateData);
+        })
+        .then(function() {
+          return findModelInstanceAsync('modelWithDeepNestedDecimalProps', instanceId);
+        })
+        .then(function(updatedInstance) {
+          updatedInstance.imdb.rating.should.be.instanceOf(Decimal128);
+          updatedInstance.imdb.rating.should.deepEqual(Decimal128.fromString('7.5'));
+          updatedInstance.imdb.innerObj.innerObj.innerObj.innerObj.testDecimal
+            .should.be.instanceOf(Decimal128);
+          updatedInstance.imdb.innerObj.innerObj.innerObj.innerObj.testDecimal
+            .should.deepEqual(Decimal128.fromString('77.77'));
+          updatedInstance.imdb.innerArray[0].testDecimal
+            .should.be.instanceOf(Decimal128);
+          updatedInstance.imdb.innerArray[0].testDecimal
+            .should.deepEqual(Decimal128.fromString('11.11'));
+          updatedInstance.imdb.innerArray[0].ObjInsideAnArray.testDecimal
+            .should.deepEqual(Decimal128.fromString('77.77'));
+          updatedInstance.imdb.innerArray[0].ObjInsideAnArray.testDecimal
+            .should.be.instanceOf(Decimal128);
+          updatedInstance.imdb.innerArray[0].nestedArray[0]
+            .testDecimal.should.be.instanceOf(Decimal128);
+          updatedInstance.imdb.innerArray[0].nestedArray[0]
+            .testDecimal.should.deepEqual(Decimal128.fromString('77.77'));
+          done();
+        });
+    });
+
+    function findModelInstance(modelName, id, cb) {
+      db.connector.execute(modelName, 'findOne', {_id: {$eq: id}}, {safe: true}, cb);
+    }
+    var findModelInstanceAsync = promisify(findModelInstance);
```

Comment:
Use `const`, not `var` - see https://loopback.io/doc/en/contrib/style-guide-es6.html#variable-declarations